src/main/java/ch/njol/skript/entity/EntityData.java

```diff
@@ -296,25 +327,80 @@ public final boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean is
+	 *              If an optional value was omitted by the user, it will still be present in the array
+	 *              with a value of {@code null}.
+	 * @param matchedPattern The index of the pattern which matched.
+	 * @param parseResult Additional information from the parser.
+	 * @return {@code true} if initialization was successful, otherwise {@code false}.
+	 */
 	protected abstract boolean init(Literal<?>[] exprs, int matchedPattern, ParseResult parseResult);
 
 	/**
+	 * Initializes this {@link EntityData} from either an entity class or a specific {@link Entity}.
+	 * <p>
+	 *     Example usage:
+	 *     	<pre>
+	 *     	    <code>
+	 *     	        spawn a pig at location(0, 0, 0):
+	 *     	        	set {_entity} to event-entity
+	 *     	        spawn {_entity} at location(0, 0, 0)
+	 *     	    </code>
+	 *     	</pre>
+	 * </p>
 	 * @param entityClass An entity's class, e.g. Player
 	 * @param entity An actual entity, or null to get an entity data for an entity class
-	 * @return Whether initialisation was successful
+	 * @return {@code true} if initialization was successful, otherwise {@code false}.
 	 */
 	protected abstract boolean init(@Nullable Class<? extends E> entityClass, @Nullable E entity);
 
+	/**
+	 * Applies this {@link EntityData} to a newly spawned {@link Entity}.
+	 * <p>
+	 *     This is used during entity spawning to set additional data, such as a saddled pig.
+	 * </p>
+	 * @param entity The spawned entity.
+	 */
 	public abstract void set(E entity);
 
+	/**
+	 * Determines whether the given {@link Entity} matches this {@link EntityData} data.
+	 * <p>
+	 *     For example:
+	 *     <pre>
+	 *         <code>
+	 *             spawn a pig at location(0, 0, 0):
+	 *             		set {_entity} to event-entity
+	 *             	if {_entity} is a pig:          # will pass
+	 *             	if {_entity} is a saddled pig:  # will not pass
+	 *         </code>
+	 *     </pre>
+	 * </p>
+	 * @param entity The {@link Entity} to match.
+	 * @return {@code true} if the entity matches, otherwise {@code false}.
+	 */
 	protected abstract boolean match(E entity);
 
+	/**
+	 * Returns the {@link Class} of the {@link Entity} that this {@link EntityData} represents or handles.
+	 *
+	 * @return The entity's {@link Class}, such as {@code Pig.class}.
+	 */
 	public abstract Class<? extends E> getType();
 
 	/**
-	 * Returns the super type of this entity data, e.g. 'wolf' for 'angry wolf'.
+	 * Returns a more general version of this {@link EntityData} with specific data removed.
+	 * <p>
+	 *     For example, calling this on {@code "a saddled pig"} would return {@code "a pig"}.
+	 *     This is typically used to obtain the base entity type without any modifiers or traits.
+	 * </p>
 	 *
-	 * @return The supertype of this entity data. Must not be null.
+	 * @return A generalized {@link EntityData} representing the base entity type. Should never return {@code null}.
```

Comment:
```suggestion
	 * @return A generalized {@link EntityData} representing the base entity type.
```
annotate @ NotNull if you want that